### PR TITLE
Fixes #28884 - DatePicker: fix calendar

### PR DESCRIPTION
**Pull request by kamils-iRonin**
image

### CODE REVIEW

**Comment by xprazak2** [comment]
I am not opposed to the change, I just could not find a page where this is broken.

Any concerns @MariaAga?

**Comment by kamils-iRonin** [comment]
> I just could not find a page where this is broken.

@xprazak2 I noticed a problem with the calendar when I wanted to use this component in the plugin

**Comment by MariaAga** [comment]
I'm also not sure how to reproduce this issue, but if we add `popoverPlacement` it needs to be added to the `TimePicker` as well I think.
Also the changes from list to div causes the `Today button` to be outside of the datepicker
Before:
image

After:
image

**Comment by kamils-iRonin** [comment]
> I'm also not sure how to reproduce this issue

@MariaAga  I added component to `componentRegistry.js`:
```javascript
import DatePicker from './common/DateTimePicker/DatePicker';
// ...
const coreComponets = [
   // ...
  { name: 'DatePicker', type: DatePicker}
]
```
and wanted to use a component in the view in this way:
```erb
<div id='datepicker'></div>
<%= mount_react_component('DatePicker', '#datepicker') %>
```
result:
image

**Comment by MariaAga** [comment]
Please note that if you want to use the component as a form you should add a function similar to this one: https://github.com/theforeman/foreman/blob/fc48ba39fabc088f3acccc94066b638cb595a8c6/app/helpers/form_helper.rb#L406
The issue in storybook is only in storybook and doesn't reproduce in Foreman.

File: webpack/assets/javascripts/react_app/components/common/DateTimePicker/DatePicker.js [hunk 1]
```diff
@@ -77,10 +77,12 @@ DatePicker.propTypes = {
   value: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string]),
   locale: PropTypes.string,
   weekStartsOn: PropTypes.number,
+  popoverPlacement: PropTypes.string,
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/common/DateTimePicker/DatePicker.js]: can you rename to `placement` like in the DateTimePicker?
https://github.com/theforeman/foreman/blob/fc48ba39fabc088f3acccc94066b638cb595a8c6/webpack/assets/javascripts/react_app/components/common/DateTimePicker/DateTimePicker.js#L126

**Comment by timogoebel** [comment]
@kamils-iRonin: Do you mind rebasing the PR?

**Review by amirfefer — COMMENTED** [review]
Thanks @kamils-iRonin , works well with no regression in foreman core, can you please add  this `DatePicker` component to our storybook (as soon as it becomes a public component (ComponentRegistery) it needs documentation)

**Review by amirfefer — APPROVED** [review]
Thanks @kamils-iRonin ! works well

**Pull request merged**